web/lib/src/helpers/extensions.dart

```diff
@@ -103,3 +103,30 @@ extension TouchListConvert on TouchList {
   @Deprecated('Use JSImmutableListWrapper<TouchList, Touch> directly instead.')
   List<Touch> toList() => JSImmutableListWrapper<TouchList, Touch>(this);
 }
+
+extension XMLHttpRequestGlue on XMLHttpRequest {
+  Map<String, String> get responseHeaders {
```

Comment:
Can we add a dartdoc comment here explaining what the output will look like? It looks like it's documented internally, so it could be as simple as reformatting that documentation. Or we could copy the `dart:html` definition:

```dart
  /// Returns all response headers as a key-value map.
  ///
  /// Multiple values for the same header key can be combined into one,
  /// separated by a comma and a space.
  ///
  /// See: http://www.w3.org/TR/XMLHttpRequest/#the-getresponseheader()-method
```